Sprint 06

Dath-Remar opened the pull request:
Выполнил задачу по спринту 6

Dath-Remar committed:
Сделал задачу по 6 спринту

Dath-Remar committed:
Адаптировал функцию вопроса про рейтинг для фильмов из топ 250

Dath-Remar committed:
Детальная информация по ошибке загрузки изображения

Dath-Remar committed:
исправлена ошибка отображения текста вопроса при загрузке

Dath-Remar committed:
отформатировал

Dath-Remar committed:
исправил повторяющийся рейтинг в вопросе(чтобы он был рандомным и не повторялся)

Nikitaaeee commented on MovieQuiz/Services/QuestionFactory.swift:
```diff
@@ -1,15 +1,87 @@
-    weak var delegate: QuestionFactoryDelegate?
+    
+    // MARK: - Public Properties
+    
+    var delegate: QuestionFactoryDelegate?
+    
```
Свойство `delegate` принято ослаблять модификатором `weak` для избежания retain cycle и утечек памяти.

Nikitaaeee commented on MovieQuiz/Services/QuestionFactory.swift:
```diff
@@ -1,15 +1,87 @@
+            } while randomRating == self.lastRandomRating
+            self.lastRandomRating = randomRating
+
+            let text = "Рейтинг этого фильма больше чем \(randomRating)?"
+            let correctAnswer = rating > randomRating
+
```
Здорово, что ты добавил рандомный рейтинг!
Если у тебя будет время и желание - можешь также добавить функционал с работой варианта с проверкой "Рейтинг этого фильма больше **меньше** чем ...", чтобы сделать более интересной (так как большая часть фильмов имеет рейтинг больше 8.1) 💡

Nikitaaeee commented on MovieQuiz/Services/QuestionFactory.swift:
```diff
@@ -50,16 +122,11 @@ final class QuestionFactory: QuestionFactoryProtocol {
-        delegate?.didReceiveNextQuestion(question: question)
+    ] */
+}
+extension Float {
+    func rounded(toPlaces places: Int) -> Float {
+        let divisor = pow(10.0, Float(places))
+        return (self * divisor).rounded() / divisor
```
Расширения на базовые типы лучше выносить в отдельные файлы.
Например, можно создать папку **Extensions** и внутри создавать по файлу для каждого типа, используя, например, такой нейминг: 
`Float+Extensions`
`UIColor+Extensions`

В каждом таком файле можно хранить все расширения к соответствующему типу 😊

Dath-Remar committed:
Исправил: Свойство delegate принято ослаблять модификатором weak для избежания retain cycle и утечек памяти.

Dath-Remar committed:
Исправил: Расширения на базовые типы лучше выносить в отдельные файлы.
Например, можно создать папку Extensions и внутри создавать по файлу для каждого типа, используя, например, такой нейминг:
Float+Extensions

Dath-Remar committed:
Исправил:Если у тебя будет время и желание - можешь также добавить функционал с работой варианта с проверкой "Рейтинг этого фильма больше меньше чем ...", чтобы сделать более интересной (так как большая часть фильмов имеет рейтинг больше 8.1) 💡

Dath-Remar committed:
Исправил: Можно попробоать добавить показ activityIndicator в момент, пока грузится следующий вопрос и его постер. + добавил анимацию перехода для изображения и текста вопроса
